### DISCUSSION TITLE
I love the work being done here, Willing to send a pr to zipline?

### DISCUSSION THREAD

**Discussion by Shabinder** [dokar3/quickjs-kt#115] [Q&A]
Hey @dokar3,

Admire the effort you put into this, I come from Zipline and am checking this out. 

I have been wanting to add windows support which is currently missing from zipline, since you have gone through the foundational work yourself already, and if willing would you be open to contribute to zipline adding windows support ?, since I am currently not that familiar with quickJS itself and the whole c/zig build process.

It would be great if that was possible, else when I may pickup, I will learn a thing or two.

The related discussion : https://github.com/cashapp/zipline/discussions/1342

**Comment by dokar3** [comment]
Thank you for the information!

I can't say if it's possible to bring Windows support to Zipline using the build process of this library. Zipline uses cklib to build the quickjs binaries, introducing zig as the build tool might be a breaking change for the team, I'm not sure they will accept this. The build process of this library is also immature and a bit hacky (mixing of `buildSrc`, `cmake`, and `zig`), which makes it harder to integrate into a company repo IMO.

I didn't use cklib from the beginning because when I started this project cklib still lacked Windows support. Since it seems to get Windows support now, it might be easier to use Zipline's existing build tool for the Windows build. Building quickjs on Windows requires MinGW and [pthread for Windows](https://sourceforge.net/projects/pthreads4w/), I'm not familiar with cklib, so maybe you can look into these and try to combine them.

**Comment by Shabinder** [comment]
@dokar3 
also fyi, the initial work to bring zipline to cross compile has been already done, see PR:
https://github.com/cashapp/zipline/pull/421

**Reply by dokar3** [reply]
That's new to me! I wonder why they haven't migrated to zig yet.
